internal/api/v1/service/delete.go

```diff
@@ -23,6 +48,76 @@ func (ctr Controller) Delete(c *gin.Context) apierror.APIErrors {
 		return err
 	}
 
+	logger.Info("getting helm client")
+
+	client, err := helm.GetHelmClient(cluster.RestConfig, logger, namespace)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	logger.Info("looking for service")
+	releaseName := names.ServiceHelmChartName(serviceName, namespace)
+	srv, err := client.GetRelease(releaseName)
+	if err != nil {
+		if errors.Is(err, helmdriver.ErrReleaseNotFound) {
+			return apierror.NewNotFoundError(fmt.Sprintf("%s - %s", err.Error(), releaseName))
+		}
+		return apierror.InternalError(err)
+	}
+
+	logger.Info(fmt.Sprintf("service found %+v\n", serviceName))
+	if srv.Info.Status != helmrelease.StatusDeployed {
+		return apierror.InternalError(err)
+	}
```

Comment:
@enrichman I added this block of code to have the `delete service` endpoint check that the instance to be deleted actually exists.

As is done by `bind` and `unbind`. After adapting the delete tests to provide the new delete request structure I consistently get `404 release not found` coming from this code.

Was it intentional to not have this validation in the delete endpoint ?
If not then I will have to dig again into the test setup to see if/why no release is present ...
And/or test cli first, as that should match on all sides.

(side note: should be factored into a helper function)

Reply:
Yes, it is definitely this piece of code tripping the test now.